### DISCUSSION TITLE
How to send "ESC" "BACKSPACE" and "RETURN" to another pane via cli ?

### DISCUSSION THREAD

**Discussion by sourproton** [wezterm/wezterm#3945] [Q&A] [answered]
I'm trying to use `wezterm cli send-text "\e\b\r"` but the escape sequences are sent as literal text, not key presses. Is there a way to achieve this with wezterm currently without setting up a `SendKey` bind in the config?

**Comment by wez** [comment]
Use the `--no-paste` flag. See https://wezfurlong.org/wezterm/cli/cli/send-text.html or `wezterm cli send-text --help` for more information

**Reply by sourproton** [reply]
Thank you for the suggestion, but it doesn't work either:
![image](https://github.com/wez/wezterm/assets/66533348/bcb6358e-09f2-4177-9b1b-cd2062777be0)

**Reply by 0f-0b** [reply, marked as answer]
Your shell most likely does not interpret escape sequences inside double quotes. If the shell [supports ANSI-C quoting](https://unix.stackexchange.com/a/371873), something like `wezterm cli send-text --no-paste $'\r'` should work.

**Reply by sourproton** [reply]
Never thought this could be the shell's problem, Thank you.

**Reply by kassio** [reply]
With this would I also be able to send something like ctrl-l? Do we have a list of these codes somewhere?

**Reply by wez** [reply]
In the standard/xterm keyboard encoding, `CTRL-<something>` can be found by looking up the ASCII value for the uppercase character, then subtracting 0x40.  So for `CTRL-L`, `L` is ASCII code `0x4c`, therefore the encoded value for `CTRL-L` is `0x0c`.

```console
$ wezterm cli send-text --no-paste $'\x0c'
```

**Reply by kassio** [reply]
Thanks @wez!
